api/config/extension_config.go

```diff
@@ -1,54 +1,43 @@
-package auth
+package config
 
 import (
 	"encoding/json"
 	"fmt"
 	"os"
 
-	"github.com/ninech/nctl/api"
-	"github.com/ninech/nctl/api/util"
-	"github.com/ninech/nctl/internal/format"
+	infrastructure "github.com/ninech/apis/infrastructure/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/clientcmd"
-	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
 const (
-	extensionKind    = "Config"
-	extensionGroup   = "nctl.nine.ch"
-	extensionVersion = "v1alpha1"
+	extensionKind        = "Config"
+	extensionGroup       = "nctl.nine.ch"
+	extensionVersion     = "v1alpha1"
+	NctlExtensionContext = "nctl"
 )
 
 var (
-	// ErrConfigNotFound describes a missing nctl config in the kubeconfig
-	ErrConfigNotFound configError = "nctl config not found"
+	// ErrExtensionConfigNotFound describes a missing nctl config in the kubeconfig
+	ErrExtensionConfigNotFound extensionConfigError = "nctl config not found"
 )
 
-type configError string
+type extensionConfigError string
 
-func (c configError) Error() string {
+func (c extensionConfigError) Error() string {
 	return string(c)
 }
 
-// IsConfigNotFoundError returns true if the nctl config could not be found in
+// IsExtensionConfigNotFoundError returns true if the nctl config could not be found in
 // the kubconfig context
-func IsConfigNotFoundError(err error) bool {
-	return err == ErrConfigNotFound
+func IsExtensionConfigNotFoundError(err error) bool {
+	return err == ErrExtensionConfigNotFound
 }
 
-// ReloginNeeded returns an error which outputs the given err with a message
-// saying that a re-login is needed.
-func ReloginNeeded(err error) error {
-	return fmt.Errorf(
-		"%w, please re-login by executing %q",
-		err,
-		format.Command().Login(),
-	)
-}
-
-// Config is used to store information in the kubeconfig context created
-type Config struct {
+// ExtensionConfig is used to store information in the kubeconfig context
+// created
+type ExtensionConfig struct {
```

Comment:
Nit: the name `config.ExtensionConfig` stutters a bit.

Reply:
renamed it to `config.Extension` and adapted the other related functions